File: python/list_resources.py
from get_session import FhirApi
from fhir.resources.bundle import Bundle
from rich import print

fhir_api = FhirApi()
bundle : Bundle = fhir_api.request_and_parse_fhir("CodeSystem", Bundle)
resources = [r.resource for r in bundle.entry]
resources.sort(key=lambda r: (r.name if r.title is None else r.title, r.version))
for r in resources:
    name = r.name if r.title is None else r.title
    print(f" - '{name}' ({r.url}, version {r.version})")

File: java/src/main/java/listresources/ListResources.java
package listresources;

import ca.uhn.fhir.rest.client.api.IGenericClient;
import fhirwrapper.FhirClientWrapper;
import org.hl7.fhir.r4.model.Bundle;
import org.hl7.fhir.r4.model.CodeSystem;

import java.util.Comparator;
import java.util.stream.Collectors;

public class ListResources {

    public static void main(String[] args) {
        IGenericClient fhirClient = FhirClientWrapper.getFhirClient();
        // standard pattern using IGenericClient for listing resources:
        // provide the class to search, then return as Bundle
        Bundle csBundle = fhirClient.search()
                .forResource(CodeSystem.class)
                .returnBundle(Bundle.class)
                .execute();
        // map the Bundle to a list of CodeSystems, since we know that only CS are
        // going to be present in the Bundle. Sort by URL and version ascending.
        csBundle.getEntry().stream().map(b -> (CodeSystem) b.getResource())
                .sorted(Comparator.comparing(CodeSystem::getUrl)
                .thenComparing(CodeSystem::getVersion)
        ).forEach(
                // print each entry to STDOUT
                entry -> System.out.printf("- CodeSystem '%s' (%s) version %s%n",
                        entry.getName(),
                        entry.getUrl(),
                        entry.getVersion())
                // Java type system ensures easy and consistent access to FHIR attributes
        );

    }
}
